src/target/msp432e4.c

```diff
@@ -0,0 +1,310 @@
+#define MSP432E4_SYS_CTRL_DID1_MASK     0xff000000U
+#define MSP432E4_SYS_CTRL_DID1_MSP432E4 0x10000000U
+
+/*
+ * Boot Config
+ *  [31] - b - lock register
+ *  [30:16] - 7fff - reserved
+ *  [15:13] - x - gpio port
+ *  [12:10] - x - gpio pin
+ *  [9] - b - gpio polarity
+ *  [8] - b - gpio enable
+ *  [7:5] - 111 - reserved
+ *  [4] - b - key select
+ *  [3:2] - 11 - reserved
+ *  [1:0] - bb - debug control
+ */
+#define MSP432E4_SYS_CTRL_BOOTCFG     (MSP432E4_SYS_CTRL_BASE + 0x01d0U)
+#define MSP432E4_SYS_CTRL_BOOTCFG_KEY (1U << 4U)
+
+/*
+ * Flash Peripheral Properties
+ *  [31] - 0 - reserved
+ *  [30] - b - prefetch buffer mode
+ *  [29] - b - flash mirror mode
+ *  [28] - b - dma flash access
+ *  [27:23] - 00000 - reserved
+ *  [22:19] - bbbb - eeprom sector size
+ *  [18:16] - bbb - flash sector size
+ *  [15:0] - xxxx - flash size
+ */
+#define MSP432E4_FLASH_PERIPH_PROP (MSP432E4_FLASH_CTRL_BASE + 0x0fc0U)
+
+/*
+ * SRAM Size
+ *  [31:16] - 0000 - reserved
+ *  [15:0] - xxxx - sram size
+ */
+#define MSP432E4_FLASH_SRAM_SIZE (MSP432E4_FLASH_CTRL_BASE + 0x0fc4U)
+
+/*
+ * Control1
+ *  [31:16] - xxxx - write key
+ *  [15:4] - 000 - reserved
+ *  [3] - b - commit
+ *  [2] - b - mass erase
+ *  [1] - b - erase sector
+ *  [0] - b - write
+ * Control2
+ *  [31:16] - xxxx - write key
+ *  [15:1] - 0000 - reserved
+ *  [0] - b - buffered flash memory write
+ */
+#define MSP432E4_FLASH_CTRL            (MSP432E4_FLASH_CTRL_BASE + 0x0008U)
+#define MSP432E4_FLASH_CTRL2           (MSP432E4_FLASH_CTRL_BASE + 0x0020U)
+#define MSP432E4_FLASH_CTRL_WRITE      (1U << 0U)
+#define MSP432E4_FLASH_CTRL_ERASE      (1U << 1U)
+#define MSP432E4_FLASH_CTRL_MASS_ERASE (1U << 2U)
+#define MSP432E4_FLASH_CTRL_COMMIT     (1U << 3U)
+
+/*
+ * Raw Interrupt Status
+ *  [31:14] - 00000 - reserved
+ *  [13] - b - program verify
+ *  [12] - 0 - reserved
+ *  [11] - b - erase verify
+ *  [10] - b - invalid data
+ *  [9] - b - pump voltage
+ *  [8:3] - 00 - reserved
+ *  [2] - b - eeprom status
+ *  [1] - b - program status
+ *  [0] - b - access status
+ */
+#define MSP432E4_FLASH_FCRIS (MSP432E4_FLASH_CTRL_BASE + 0x000cU)
+
+/*
+ * Flash Write Key
+ *  [31:16] - 0000 - reserved
+ *  [15:0] - xxxx - key
+ */
+#define MSP432E4_FLASH_FLPEKEY (MSP432E4_FLASH_CTRL_BASE + 0x003cU)
+
+/*
+ * Flash Access Address
+ *  [31:20] - 000 - reserved
+ *  [19:0] - xxxxx - operation aligned address
+ */
+#define MSP432E4_FLASH_ADDR (MSP432E4_FLASH_CTRL_BASE + 0x0000U)
+
+/* Flash Data */
+#define MSP432E4_FLASH_DATA (MSP432E4_FLASH_CTRL_BASE + 0x0004U)
+
+/*
+ * EEPROM Size
+ *  [31:16] - xxxx - # 16bit words
+ *  [15:0] - xxxx - # 32bit words
+ */
+#define MSP432E4_EEPROM_SIZE (MSP432E4_EEPROM_BASE + 0x0000U)
+
+#define MSP432E4_SRAM_BASE  0x20000000U
+#define MSP432E4_FLASH_BASE 0x00000000U
+
+/*
+ * XXX: The flash block size might be nice to use, but the part used in
+ * testing has a 16k sector which is too big w/ the current build setup.
+ */
+#define BUFFERED_WRITE_SIZE 0x100
```

Comment:
See comment below.

I would like to see a comment about the *actual* writesize and why it's not used, as the current comment is not strictly correct, the problem is not that the sector size is too big, but that the write size is too small